src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -6566,6 +6590,15 @@ private void ReportNullabilityMismatchWithTargetDelegate(Location location, Name
             }
 
             Debug.Assert(delegateType is object);
+
+            if (unboundLambda.HasExplicitReturnType(out _, out var returnType) &&
+                IsNullabilityMismatch(invoke.ReturnTypeWithAnnotations, returnType, requireIdentity: true))
```

Comment:
Should this be `requireIdentity: false` instead?

Reply:
Requiring the return type nullability to match exactly corresponds to the existing requirement for parameters. We could relax this requirement but we should probably change the requirement for both parameters and return types at the same time.